modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/context/index.js

```diff
@@ -28,11 +32,33 @@ const AppContextProvider = ({assetsPath, children, page}) => {
 
 	const userAccount = data?.userAccount;
 
+	useEffect(() => {
+		const getSessionId = async () => {
+			const session = await fetchSession(oktaSessionURL);
+
+			if (session) {
+				dispatch({
+					payload: session.id,
+					type: actionTypes.UPDATE_SESSION_ID,
+				});
+			}
+		};
+
+		getSessionId();
+	}, [oktaSessionURL]);
+
 	useEffect(() => {
 		const projectExternalReferenceCode = SearchParams.get(
 			PARAMS_KEYS.PROJECT_APPLICATION_EXTERNAL_REFERENCE_CODE
 		);
 
+		window.addEventListener('customer-portal-menu-selected', ({detail}) => {
```

Comment:
This event listener should be removed, in the call back of useEffect

Reply:
I'll be adding that! Thanks